### PR TITLE
Requirements and roles description cleanup

### PR DESCRIPTION
**Pull request by sbesson**
This PR:

- fixes the top-level `requirements.yml` to consume all roles from Galaxy following the resolution of https://github.com/ansible/galaxy-issues/issues/259
- updates the `ansible/README.md` to remove the hard-coded list of roles in favor of the dynamic Galaxy openmicroscopy page only
- adds a `ansible/roles/README.md` file explaining the short-term strategy for this directory (see also https://github.com/openmicroscopy/infrastructure/pull/272#issuecomment-303366867)

### CODE REVIEW

File: ansible/README.md [hunk 1]
```diff
@@ -17,75 +17,21 @@ For more explanation on using Ansible in general, you may want to read:
...
- - omero-user
-   ([GitHub](https://github.com/openmicroscopy/ansible-role-omero-user)):
-   Create OMERO user accounts and groups
- - omero-web-apps
-   ([GitHub](https://github.com/openmicroscopy/ansible-role-omero-web-apps)):
-   Install OMERO.web plugins
- - reboot-server
-   ([GitHub](https://github.com/openmicroscopy/ansible-role-reboot-server)):
-   Reboot a server, optionally wait for it to return.
+Core roles can be found under the
```
Comment (joshmoore) [inline ansible/README.md]: Worth a screenshot or will that be confusing?

Comment (sbesson) [inline ansible/README.md]: See https://github.com/sbesson/infrastructure/tree/roles_cleanup/ansible for the rendering of the last commit  6b04d4f

File: ansible/README.md [hunk 2]
```diff
@@ -17,75 +17,31 @@ For more explanation on using Ansible in general, you may want to read:
...
-
 Roles
 -----
 
-Roles are "the Ansible way of bundling automation content and making it reusable." To keep playbooks
-as simple as possible, logic is refactored out into roles and versioned separately. A list of most
-of the roles can be found in
+Roles are "the Ansible way of bundling automation content and making it
+reusable." To keep playbooks as simple as possible, logic is refactored out
+into roles, maintaining in individual repositories and versioned separately. A
```
Comment (mtbc) [inline ansible/README.md]: "maintained"

File: ansible/README.md [hunk 3]
```diff
@@ -17,75 +17,31 @@ For more explanation on using Ansible in general, you may want to read:
...
-   Create OMERO user accounts and groups
- - omero-web-apps
-   ([GitHub](https://github.com/openmicroscopy/ansible-role-omero-web-apps)):
-   Install OMERO.web plugins
- - reboot-server
-   ([GitHub](https://github.com/openmicroscopy/ansible-role-reboot-server)):
-   Reboot a server, optionally wait for it to return.
+Core roles can be found under the
+[openmicroscopy](https://galaxy.ansible.com/openmicroscopy/) account on Galaxy
+and are versioned using [Semantic Versioning](http://semver.org/). Roles which
```
Comment (mtbc) [inline ansible/README.md]: s/which/whose/ perhaps

**Comment by manics** [comment]
Using https://github.com/snoopycrimecop/infrastructure/commit/29a45a1a74cfdefac8355f17b4fe3ba6befe5334

Can you add an `openmicroscopy.` prefix to `logrotate`:
```
$ ansible-playbook -i ../../ansible/inventory/idr-hosts idr-openstack-post.yml --list-hosts
ERROR! the role 'logrotate' was not found ... 
The error appears to have been in '... infrastructure/ansible/idr-openstack-post.yml': line 10, column 5, but may be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

  - role: network
  - role: logrotate
    ^ here
```

Also the first section of the main README "This folder contains a variety of Ansible playbooks ..." is out of date, but this can be left for a later PR.

**Comment by sbesson** [comment]
@manics pushed

**Comment by manics** [comment]
and same for `system-monitor-agent`... after making that change locally `ansible-playbook -i ../../ansible/inventory/idr-hosts idr-openstack-post.yml -C --ask-become-pass` works

**Pull request merged**